corehq/motech/repeaters/views/repeat_records.py

```diff
@@ -394,21 +396,8 @@ def _change_record_state(base_string, string_to_add):
 
 
 def _url_parameters_to_dict(url_params):
-    dict_to_return = {}
-    if not url_params:
-        return dict_to_return
-
-    while url_params != '':
-        pos_one = url_params.find('=')
-        pos_two = url_params.find('&')
-        if pos_two == -1:
-            pos_two = len(url_params)
-        key = url_params[:pos_one]
-        value = url_params[pos_one+1:pos_two]
-        dict_to_return[key] = value
-        url_params = url_params[pos_two+1:] if pos_two != len(url_params) else ''
-
-    return dict_to_return
+    name_value_pairs = parse_qsl(url_params, keep_blank_values=True)
```

Comment:
is there a reason you don't use `parse_qs`? I think that returns a dict right away.

Reply:
`parse_qs()` returns a dict of lists. The code expects to get scalar values.
```python
>>> qs = 'foo=1&foo=2&bar=3'
>>> parse_qs(qs)
{'foo': ['1', '2'], 'bar': ['3']}
>>> dict(parse_qsl(qs))
{'foo': '2', 'bar': '3'}
```
I agree that generally `parse_qs()` is what you want, but I wanted to change this code's behaviour as little as possible.